### ISSUE TITLE
Application in Container with no Exposed Ports fail

### ISSUE THREAD

**Issue by heynemann** [deis/workflow#617]
Hey everyone,

I'm trying to deploy an app that's not HTTP and has no exposed ports (it's just a worker). The way I deploy it is using docker, thus I do `deis pull -a khan-prune tfgco/khan-prune:latest`.

The first time I tried using this command I got: 
```
Creating build... Error: Unknown Error (400): {"detail":"PORT needs to be set in the application config when using a private registry"}
```

Every time I use this command subsequently after setting an arbitrary PORT (8000), I got this result:
```
Creating build... Error: Unknown Error (400): {"detail":"(app::deploy): No build associated with this release"}
```

The worst thing here is that the app got to Kube and IS RUNNING FINE!  I can see by its logs.

In Kubernetes I can see that the problem very likely lies with an issue with the Readiness Probe since this is the message I see in the events section for my app replica sets:

```
Readiness probe failed: dial tcp 100.96.5.246:8000: getsockopt: connection refused
```

If someone wants to reproduce this, the container for khan-prune is public and published at docker-hub.

Thanks for any help!

**Comment by kmala** [comment]
this should fix your issue https://github.com/deis/controller/pull/1136

**Comment by bacongobbler** [comment]
Note that ^^ has not made into a release yet, however we are planning on cutting v2.9.0 next week. You can fork the controller and use that patch on your existing installation for the time being, however.

**Comment by kmala** [comment]
This should be resolved by installing the 2.9.0 workflow.
